src/xrc/src/api/test.rs

```diff
@@ -587,6 +587,25 @@ fn get_exchange_rate_for_crypto_non_usd_pair() {
                             quote_asset_num_queried_sources: 4,
                             quote_asset_num_received_rates: 4,
                             forex_timestamp: Some(0),
+                        },
+                    // It is necessary to have a CXDR rate with at least 4 sources for the rate store to return a result
+                    "CXDR".to_string() =>
+                        QueriedExchangeRate {
```

Comment:
The same `QueriedExchangeRate` struct appears many times. How about defining it once and reusing it?

Also, it is recommended to use the constructor to create an instance whenever possible.   :-)